src/helper_functions.jl

```diff
@@ -1,6 +1,30 @@
 # default tolerance for matrix condition number (see 'isinvertible')
 const DEFAULT_COND_TOL = 1e6
 
+"""
+    dot_zero(x::AbstractVector{N}, y::AbstractVector{N}) where{N<:Real}
+
+Dot product with preference for zero value in the presence of infinity values.
+
+### Input
+
+- `x` -- first vector
+- `y` -- second vector
+
+### Output
+
+The dot product of `x` and `y`, but with the rule that `0 * Inf == 0`.
```

Comment:
Perhaps add a short note about where is this function practically used, to not scare mathematicians ;p  (see some arguments about why this operation should be undefined [here](https://math.stackexchange.com/questions/28940/why-is-infty-cdot-0-not-clearly-equal-to-0))